test/crawler_web/controllers/keyword_controller_test.exs

```diff
@@ -0,0 +1,84 @@
+defmodule CrawlerWeb.KeywordControllerTest do
+  use CrawlerWeb.ConnCase, async: true
+
+  alias Crawler.Keyword.Keywords
+
+  describe "POST /keyword" do
+    test "given valid keyword csv file, creates keyword successfully", %{conn: conn} do
+      user = insert(:user)
+      uploaded_file = keyword_file_fixture("valid.csv")
+
+      conn =
+        conn
+        |> log_in_user(user)
+        |> post(Routes.keyword_path(conn, :create), %{keyword_csv_file: %{file: uploaded_file}})
+
+      assert get_flash(conn, :info) == gettext("Keywords were uploaded!")
+
+      keywords = Keywords.list_keywords(user.id)
+
+      list =
+        Enum.map(keywords, fn keyword ->
+          keyword.name
+        end)
+
+      assert list == ["one", "two", "three"]
```

Comment:
A small comment here

1.

We don't really need 

```
      list =
        Enum.map(keywords, fn keyword ->
          keyword.name
        end)
```

In Elixir, I think we can do something like

```
[%Keyword{name: "one"}, %Keyword{name: "two"}, %Keyword{name: "three"}] = keywords
```

2.

As we don't really control the `return ordering` there, we don't know the one or two or three are returned first, so assessing following the order is not a safe solution as the test could be failed randomly,

You can check the issue on part 2`Database Ordering` of this blog https://nimblehq.co/blog/fast-and-stable-test-suite

Here, my suggestion is

- check the length 
- check the value in the resut

```
assert length(keywords) == 2

assert "one" in keywords.names # just an idea here
assert "two" in keywords.names # just an idea here
assert "three" in keywords.names # just an idea here
```

Reply:
Updated in [1d816b2](https://github.com/vnntsu/elixir-ic/pull/36/commits/1d816b234f96b49e84b553a585e8119e4990ed31)

Reply:
added an comment there 👍